### DISCUSSION TITLE
'artifacts' contains the ipa file path, but does not provide downloading

### DISCUSSION THREAD

**Discussion by q962** [codemagic-ci-cd/codemagic-docs#2506] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Other - please mention in your question

### Steps to reproduce

```yaml
    scripts: ...
    artifacts: 
      - app.ipa
```

### Expected results

The exported zip file contains the ipa file.

### Actual results

> apk, aab, aar, ipa, app, pkg, proguard mapping (mapping.txt), flutter_drive.log, jar, zip, xarchive and dSYM.zip files will be available as separate items in the Artifacts section on the build page. The rest of the artifacts will be included in an archive with the following name pattern: {project-name}_{version}_artifacts.zip.

Does this mean that the above files will be excluded?

### Build id (optional)

663588873d1a624ac0fe18de

**Comment by himesh-cm** [comment]
Hey @q962, You need to specify the complete path to the IPA file under the artefact section, and those files will be available for download on your build details page. You can check our sample yaml file [here](https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/ionic/ionic-capacitor-demo-project/codemagic.yaml#L53) for your reference.

Let me know if you still need assistance with your query.

**Reply by q962** [reply]
It is exactly what I have done. 

Although my zip command is missing the -r parameter. 
But the corresponding ipa file should exist.

I specify the following path for export:
```yaml
      - mv ios/App/archive_output.xcarchive/Products/{Applications,Payload}
      - zip app.ipa ios/App/archive_output.xcarchive/Products/Payload 
      - zip ios/App/app.ipa ios/App/archive_output.xcarchive/Products/Payload 
    artifacts: 
      - app.ipa
      - ios/App/archive_output.xcarchive/Products/Payload
      - ios/App/app.ipa
```

Only the second path succeeds

**Reply by himesh-cm** [reply]
Hey @q962,

Apologies for the delayed reply; the thread got missed. All the configuration and path look okay, but for the final check, I recommend that you `ssh` into the build machine as shown [here](https://docs.codemagic.io/troubleshooting/accessing-builder-machine-via-ssh/) and make sure that the artifacts do exist at the specified path.

**Reply by q962** [reply]
I have now obtained the ipa file.
I couldn't get it before, probably because the corresponding file didn't exist.
